### PR TITLE
[`SAM`] Add sam doc

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?
 
As suggested by @LysandreJik offline, this PR adds a nice docstring for `SamModel` showing users how to leverage Auto API to run SAM

### CODE REVIEW

File: src/transformers/models/sam/modeling_sam.py [hunk 1]
```diff
@@ -1270,6 +1270,34 @@ def forward(
...
+        >>> processor = AutoProcessor.from_pretrained("facebook/sam-vit-base")
+
+        >>> img_url = "https://huggingface.co/ybelkada/segment-anything/resolve/main/assets/car.png"
+        >>> raw_image = Image.open(requests.get(img_url, stream=True).raw).convert("RGB")
+        >>> input_points = [[[400, 650]]]  # 2D location of a window on the car
+        >>> inputs = processor(images=raw_image, input_points=input_points, return_tensors="pt")
+
+        >>> # Get segmentation mask
+        >>> outputs = model(**inputs)
+        >>> iou_scores = outputs["iou_scores"]
```
Comment (ArthurZucker) [inline src/transformers/models/sam/modeling_sam.py:1293]: they are not used and you don't expect them so can be removed

**Review by ArthurZucker — APPROVED** [review]
Good to me! thanks for adding

File: src/transformers/models/sam/modeling_sam.py (hunk 1 above)
Comment (younesbelkada) [reply src/transformers/models/sam/modeling_sam.py:1293]: Makes sense!

File: src/transformers/models/sam/modeling_sam.py [hunk 2]
```diff
@@ -1270,6 +1270,31 @@ def forward(
...
+
+        ```python
+        >>> from PIL import Image
+        >>> import requests
+        >>> from transformers import AutoModel, AutoProcessor
+
+        >>> model = AutoModel.from_pretrained("facebook/sam-vit-base")
+        >>> processor = AutoProcessor.from_pretrained("facebook/sam-vit-base")
+
+        >>> img_url = "https://huggingface.co/ybelkada/segment-anything/resolve/main/assets/car.png"
```
Comment (amyeroberts) [inline src/transformers/models/sam/modeling_sam.py:1284]: The image should be moved out of a personal account into an internal one

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding! 

Just a question on device placement calls

File: src/transformers/models/sam/modeling_sam.py [hunk 3]
```diff
@@ -1270,6 +1270,31 @@ def forward(
...
+        >>> raw_image = Image.open(requests.get(img_url, stream=True).raw).convert("RGB")
+        >>> input_points = [[[400, 650]]]  # 2D location of a window on the car
+        >>> inputs = processor(images=raw_image, input_points=input_points, return_tensors="pt")
+
+        >>> # Get segmentation mask
+        >>> outputs = model(**inputs)
+
+        >>> # Postprocess masks
+        >>> masks = processor.post_process_masks(
+        ...     outputs.pred_masks.cpu(), inputs["original_sizes"].cpu(), inputs["reshaped_input_sizes"].cpu()
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply src/transformers/models/sam/modeling_sam.py:1294]: not needed indeed!

File: src/transformers/models/sam/modeling_sam.py (hunk 2 above)
Comment (younesbelkada) [reply src/transformers/models/sam/modeling_sam.py:1284]: Awesome ok, will move it

Comment (younesbelkada) [reply src/transformers/models/sam/modeling_sam.py]: 3863c2e

File: src/transformers/models/sam/modeling_sam.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (younesbelkada) [reply src/transformers/models/sam/modeling_sam.py]: 3863c2e

**Pull request merged**